make adaptive expression library compatible with IE

atishayv opened the pull request:
##Fixes:
Make adaptive expression library compatible with IE

## Description
There was no support of IE in the adaptive expression library. Using babel to transpile compiled code and add polyfills so that it works in IE also.

## Specific Changes
- Using babel to transpile adaptive-expression and other dependent libraries which are not compatible with IE.
- Adding core js to resolve all polyfills which are used in those dependent libraries.
- Updated templating library package json to create a webpack build in dist folder which is compatible with IE


## Testing
Tested manually by taking dependecy in adaptive card designer in IE to make sure that there are no errors

atishayv commented:
CC: @dclaux, @jaisanth

stevengum commented on libraries/adaptive-expressions/webpack.config.js:
```diff
@@ -0,0 +1,54 @@
+        mode: 'none',
+
+        entry: {
+            index: './src/index.ts',
```
Should this be `index: path.resolve(__dirname, './src/index.ts')`?

stevengum commented on libraries/adaptive-expressions/webpack.config.js:
```diff
@@ -0,0 +1,54 @@
+                            target: 'es5',
+                        },
+                    },
+                    exclude: /node_modules/,
```
I believe a safe-list would be better than instead of excluding `node_modules`. The entry is already `'./src/index.ts'` and a safe-list provides a better "locked-down" mechanism that ensures we only ship and webpack what we mean to.
```suggestion
                    include: [
                        path.resolve(__dirname, './src')
                    ],
```

stevengum commented on libraries/adaptive-expressions/babel.config.js:
```diff
@@ -0,0 +1,26 @@
+            },
+        ],
+    ];
+    const plugins = [
+        ['@babel/plugin-proposal-decorators', { decoratorsBeforeExport: true }],
+        ['@babel/plugin-proposal-class-properties'],
+        ['@babel/plugin-transform-modules-commonjs'],
+        ['@babel/plugin-transform-runtime', { useESModules: false }],
+    ];
```
This is toolchain config and not consumed by end-users, but it would be great if you added some comments explaining what's going on here for future maintainers who may not be familiar with the current scenario and/or babel ecosystem.

atishayv replied:
Added comments . Let me know if its okay

joshgummersall commented on libraries/adaptive-expressions-ie11/webpack.config.js:
```diff
@@ -0,0 +1,44 @@
+module.exports = (env, argv) => {
+    return {
+        mode: 'none',
+        entry: ['core-js/modules/es.array.find', path.resolve(__dirname, './src/index.ts')],
```
Now that this is a separate package let's put the polyfill in the entry point directly.

atishayv replied:
Actualy we dont need that polyfill now as that was added because adaptive expression library was using Array.find before.However, now we are passing adaptive-expression also to babel which has core js built in as usage, so it will autmatically get added. 
Nice catch :-)

joshgummersall reviewed (changes requested):
One small comment, can you verify that this bundle still works for your use case as well?

atishayv commented:
> One small comment, can you verify that this bundle still works for your use case as well?

Yes, I did verify it with adaptive-designer. I had to add aliases for the same but it worked. These are the 3 changes I did : 
1) package json, I removed the expression dependency and added dependency for ie 11 : 
![image](https://user-images.githubusercontent.com/13940020/96601065-45b72600-130f-11eb-91c8-f428e724b45e.png)
2) Added alias in webpack:
![image](https://user-images.githubusercontent.com/13940020/96601196-667f7b80-130f-11eb-8d45-d197475aae98.png)
3) Added alias in tsconfig:
![image](https://user-images.githubusercontent.com/13940020/96601281-7a2ae200-130f-11eb-84a3-d78b30a3c751.png)

CC : @dclaux, @jaisanth

atishayv commented:
Sorry, by mistake i closed the PR. Have reopened it

joshgummersall reviewed (approved):
Looks good to me! Thanks for your work on this 👍

atishayv commented:
> Looks good to me! Thanks for your work on this 👍

Can we merge this ?

stevengum commented on libraries/adaptive-expressions-ie11/src/index.ts:
```diff
@@ -0,0 +1 @@
+export * from 'adaptive-expressions';
```
I feel a bit silly for saying this, but can you add the appropriate copyright headers to the file?

```suggestion
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

export * from 'adaptive-expressions';
```

stevengum commented on libraries/adaptive-expressions-ie11/package.json:
```diff
@@ -0,0 +1,50 @@
+    "set-version": "npm version --allow-same-version ${Version}"
+  },
+  "files": [
+    "src",
```
Is there any benefit to us including `"src"`? If someone wanted to use the `"src"`, they should likely instead use `adaptive-expressions`, right?

atishayv replied:
Yes, thanks for pushing those changes

stevengum commented on libraries/adaptive-expressions-ie11/package.json:
```diff
@@ -0,0 +1,50 @@
+    "set-version": "npm version --allow-same-version ${Version}"
+  },
+  "files": [
+    "src",
```
```suggestion
```
